tools/qontract_cli.py

```diff
@@ -491,6 +491,57 @@ def service_owners_for_rds_instance(ctx, aws_account, identifier):
     print_output(ctx.obj['output'], service_owners, columns)
 
 
+@get.command()
+@click.pass_context
+def sre_checkpoints(ctx):
+    sre_checkpoints_raw = queries.get_sre_checkpoints()
+    apps = queries.get_apps()
+
+    def full_name(app):
+        name = app['name']
+        if app.get('parentApp'):
+            parent_app = app['parentApp']['name']
+            name = f"{parent_app}/{name}"
+        return name
+
+    sre_checkpoints = {}
+    for checkpoint in sre_checkpoints_raw:
+        app = checkpoint['app']
+        name = full_name(app)
+
+        if name not in sre_checkpoints:
+            sre_checkpoints[name] = checkpoint['date']
+        else:
+            if checkpoint['date'] > sre_checkpoints[name]:
+                sre_checkpoints[name] = checkpoint['date']
+
+    parent_apps = {
+        app['parentApp']['path']
+        for app in apps
+        if app.get('parentApp')
+    }
+
+    app_checkpoints = []
+    for app in apps:
+        # skip parent apps
+        if app['path'] in parent_apps:
+            continue
+
+        # consider only onboardedApps
+        if app['onboardingStatus'] != 'OnBoarded':
+            continue
```

Comment:
Shorter:

```python
if app['path'] not in parent_apps and app['onboardingStatus'] == 'OnBoarded':
    app_checkpoints.append(...)
```
which may lend itself to a list comprehension.

Reply:
In general, in terms of legibility, do you prefer your way? I think in terms of performance is the same, and with regard to legibility, I believe I would understand the separated if clauses better than a long if. Also the long if holds because there are just two terms, but for more terms I really prefer to separate them.

Reply:
Yeah, I find a single `and` expression much easier to understand than a string of if-continues. Also, if this were a complicated expression, I doubt 10 2-line blocks, spanning 20 lines would be easier to read than a single if with 10 conditions.